File: M/1798-maximum-number-of-consecutive-values-you-can-make/main.go
package main

import (
	"fmt"
	"sort"
)

func getMaximumConsecutive(coins []int) int {
	sum := 0
	sort.Ints(coins)

	for _, c := range coins {
		if sum+1 < c {
			return sum + 1
		}
		sum += c
	}
	return sum + 1
}

func main() {
	fmt.Println(getMaximumConsecutive([]int{1, 4, 10, 3, 1}))
}

File: M/1798-maximum-number-of-consecutive-values-you-can-make/main.rs
struct Solution {}

impl Solution {
  pub fn get_maximum_consecutive(coins: Vec<i32>) -> i32 {
    let mut max: i32 = 0;

    let mut cc: Vec<i32> = coins;
    cc.sort();

    for c in cc {
      if max + 1 < c {
        return max + 1;
      }
      max += c;
    }
    max + 1
  }
}
